src/raftstore/store/peer.rs

```diff
@@ -485,6 +495,43 @@ impl Peer {
         StaleState::Valid
     }
 
+    fn next_lease_expired_time(&self, send_to_quorum_ts: Timespec) -> Timespec {
+        // The valid leader lease should be
+        // "lease = election_timeout - (quorum_commit_ts - send_to_quorum_ts)"
+        // And the expired timestamp for that leader lease is "quorum_commit_ts + lease",
+        // which is "send_to_quorum_ts + election_timeout" in short.
+        send_to_quorum_ts + self.election_timeout
+    }
+
+    fn update_leader_lease(&mut self, ready: &Ready) {
+        // Update leader lease when the raft state changes
+        if ready.ss.is_some() {
+            let ss = ready.ss.as_ref().unwrap();
+            match ss.raft_state {
+                StateRole::Leader => {
+                    // The local read could only be performed after a new leader has applied
+                    // the first empty entry on its term. After that the lease expired time
+                    // should be updated to
+                    //   send_to_quorum_ts + election_timeout
+                    // as the function `next_lease_expired_time` explains.
+                    // The lease expired time could be updated ahead right after it becomes
+                    // leader, since it's more convenient to do it here and it does not
+                    // hurt the correctness.
```

Comment:
It is recommended to update the lease expiring time right after the leader is applied because it's more convenient to do it here and it has no impact on the correctness.